cmd/otelcol/main.go

```diff
@@ -108,26 +108,32 @@ func main() {
 		)
 	}
 
-	emp := envprovider.New()
-	fmp := fileprovider.New()
+	configServer := configconverter.NewConfigServer()
+	configMapConverters = append(configMapConverters, configServer)
+
+	envProvider := envprovider.New()
+	fileProvider := fileprovider.New()
 	serviceConfigProvider, err := service.NewConfigProvider(
 		service.ConfigProviderSettings{
-			Locations: configLocations(inputFlags),
-			MapProviders: map[string]confmap.Provider{
-				emp.Scheme(): configprovider.NewConfigSourceConfigMapProvider(
-					emp,
-					zap.NewNop(), // The service logger is not available yet, setting it to NoP.
-					info,
-					configsources.Get()...,
-				),
-				fmp.Scheme(): configprovider.NewConfigSourceConfigMapProvider(
-					fmp,
-					zap.NewNop(), // The service logger is not available yet, setting it to NoP.
-					info,
-					configsources.Get()...,
-				),
+			ResolverSettings: confmap.ResolverSettings{
+				URIs: configLocations(inputFlags),
+				Providers: map[string]confmap.Provider{
+					envProvider.Scheme(): configprovider.NewConfigSourceConfigMapProvider(
+						envProvider,
+						zap.NewNop(), // The service logger is not available yet, setting it to NoP.
```

Comment:
this gives me a bit of a pause - is it set later? Is it important? It looks like at minimum we might want to fix this in a subsequent PR.

Reply:
This inherits the existing functionality, but deferred service logger creation is a problem in our distribution as a whole: https://github.com/open-telemetry/opentelemetry-collector/blob/main/service/service.go#L67.